x-team/xp-ui
# Create work experience editor

**Release Type:** *Non-Breaking feature* 

Fixes https://x-team-internal.atlassian.net/browse/XP-3736

## Description

Implement a `WorkExperienceEntryEditor` component, which allows for condensed editing of work experience entries. This component is intended for integration in the XP Admin dashboard, and so is briefer and has less extensive validation than the applicant-facing work experience page. However, many of the same components are reused.

## Checklist


## Steps to Test or Reproduce

Open Storybook and navigate to UI Components > Work Experience > Editor. This will show you the edit view on the work experience form. Validate that you can edit and work with all controls as expected.

In UI Components > Work Experience > "basic usage" you can test the combined Presenter/Editor function. The storybook has a stub implementation that allows you to edit an experience entry inline and then save it, and you will see new entries added to the list. You can also click the "Add New Experience" button at the bottom of the list to add an entirely new entry.

You may also look at the console and observe the object that is passed to the `onSave` method after you click "Save". The storybook implementation logs this object to the console so that you can validate that all modified fields are passed to the `onSave` method.

Note that in the "No experience provided" case, you can't actually click the "Add New Experience" button and have your newly-created experience be added. This is a shortcoming of the Storybook stub, not the control itself.

## Impacted Areas in Application

XP Admin interface.

## Screenshots

![image](https://user-images.githubusercontent.com/5854132/81071495-79cb9580-8eed-11ea-9457-fd0b8ea78ff6.png)
